autonomy/cli/scaffold_fsm.py

```diff
@@ -753,6 +756,193 @@ def _add_abstract_round_abci_if_not_present(ctx: Context) -> None:
         add_item(ctx, SKILL, abstract_round_abci_public_id)
 
 
+# Scaffolding of tests
+class RoundTestsFileGenerator(RoundFileGenerator):
+    """RoundTestsFileGenerator"""
+
+    FILENAME = "tests_" + ROUNDS_FILENAME
+
+    ROUNDS_FILE_HEADER = dedent(
+        """\
+        \"\"\"This package contains the tests for rounds of {FSMName}.\"\"\"
+
+        from typing import Any, FrozenSet, Hashable, Type, cast
+        from unittest import mock
+
+        import pytest
+
+        # TODO: define and import specific payloads explicitly by name 
+        from packages.{author}.skills.{skill_name}.payloads import *
+        from packages.{author}.skills.{skill_name}.rounds import (
+            Event,
+            SynchronizedData,
+            {non_degenerate_rounds}
+        )
+        from packages.valory.skills.abstract_round_abci.base import (
+            AbciAppDB,
+            AbstractRound,
+            BaseTxPayload,
+            ConsensusParams,
+        )
+
+
+        MAX_PARTICIPANTS: int = 4
+
+    """
+    )
+
+    BASE_CLASS = dedent(
+        """\
+        class BaseRoundTestClass:
```

Comment:
This makes no sense. You should use existing test tools: `FSMBehaviourBaseCase` for behaviours

and 

for rounds:

```
from tests.test_skills.test_abstract_round_abci.test_base_rounds import (
    BaseCollectDifferentUntilThresholdRoundTest,
    BaseCollectSameUntilThresholdRoundTest,
)
```
(currently, this might look weird as `tests` is not accessible from the skill. But worry not, this will live inside `abstract_round_abci` skill by the end of the week as part of our test refactor.